packages/web-api/src/file-upload.ts

```diff
@@ -0,0 +1,181 @@
+import { Logger } from '@slack/logger';
+import { readFileSync, ReadStream } from 'fs';
+import { errorWithCode, ErrorCode } from './errors';
+import { FilesUploadV2Arguments, FileUploadV2, FileUploadV2Entry } from './methods';
+
+/**
+ * Returns a fileUploadEntry used internally to store state of the file upload
+ * @param options Options provided by user
+ * @param channelId optional channel id to share file with, omitted, channel is private
+ * @returns
+*/
+export async function getFileUpload(
+  options: FilesUploadV2Arguments | FileUploadV2,
+  logger: Logger, channelId?: string,
+): Promise<FileUploadV2Entry> {
+  warnIfLegacyFileType(options, logger);
+  const fileName = warnIfMissingOrInvalidFileName(options, logger);
+  const fileData = await getFileData(options);
+  const fileDataBytesLength = getFileDataLength(fileData);
+
+  const fileUploadEntry = {
+    // supplied by user
+    alt_text: options.alt_text,
+    channel_id: channelId,
+    content: options.content,
+    file: options.file,
+    filename: options.filename ?? fileName,
+    initial_comment: options.initial_comment,
+    snippet_type: options.snippet_type,
+    thread_ts: options.thread_ts,
+    title: options.title,
```

Comment:
For v1 compatibility:
```suggestion
    title: options.title ?? (options.filename ?? fileName),
```